src/arch/x86.rs

```diff
@@ -54,13 +54,22 @@ pub(super) fn dynamic_table_addr() -> *const Elf_Dyn {
         asm!(
             ".weak _DYNAMIC",
             ".hidden _DYNAMIC",
-            "call 0f",
+            "call 2f",
             ".cfi_adjust_cfa_offset 4",
-            "0:",
+            "2:",
             "pop {0}",
             ".cfi_adjust_cfa_offset -4",
-            "1:",
-            "add {0}, offset _GLOBAL_OFFSET_TABLE_+(1b-0b)",
+            "3:",
+            // Use "2" and "3" instead of "0" and "1" because "0b" and "1b" are
+            // parsed as binary literals rather than as label references. And,
+            // use a `.set` here because the assembler doesn't seem to support
+            // the symbol difference expression in an instruction operand
+            // context. Then, using a `.set` requires a symbol name, and there
+            // appears to be no syntax or convention for temporary labels that
+            // can be used with `.set`, so we use a prefix of "usercode_" to
+            // hopefully avoid colliding with any compiler symbols.
+            ".set usercode_offset, _GLOBAL_OFFSET_TABLE_+(3b-2b)",
+            "add {0}, offset usercode_offset",
```

Comment:
This adds the wrong integer to eax. Locally I'm getting the following disassembly:

```
   0xf7fb33d0 <+0>:     sub    esp,0xc
   0xf7fb33d3 <+3>:     call   0xf7fb33d8 <_ZN6origin4arch18dynamic_table_addr17h70b2129cfeed48efE+8>
   0xf7fb33d8 <+8>:     pop    eax
   0xf7fb33d9 <+9>:     add    eax,0x59f7d
=> 0xf7fb33df <+15>:    lea    eax,[eax-0x98]
   0xf7fb33e5 <+21>:    mov    DWORD PTR [esp+0x8],eax
   0xf7fb33e9 <+25>:    mov    eax,DWORD PTR [esp+0x8]
   0xf7fb33ed <+29>:    add    esp,0xc
   0xf7fb33f0 <+32>:    ret
```

At the indicated location eax should point to `_GLOBAL_OFFSET_TABLE_`, but it has 0xf800d355 as value, despite `_GLOBAL_OFFSET_TABLE_` being at 0xf7ffcf7c.

Reply:
With the second suggestion of mine applied to both this function and `ehdr_addr` all tests pass for me.